src/doc/DocAnnotator.js

```diff
@@ -496,7 +496,10 @@ class DocAnnotator extends Annotator {
         }
 
         this.createHighlightDialog.hide();
-        document.getSelection().removeAllRanges();
+
+        if (!this.isCreatingHighlight) {
+            document.getSelection().removeAllRanges();
```

Comment:
Isn't `removeAllRanges()` used in other places? Maybe abstract to a util function?

Reply:
Also, why are we calling `resetHighlightSelection` if we're already creating a highlight?

Reply:
This is the only place we're checking if you're in the middle of creating another highlight so i'm not sure if it's worth making a util function.

Reply:
This method gets called on `mouseup` and we we're only clearing the range if 1) the create dialog is already visible and 2) you're highlighting new text. Otherwise when this method would get called the create dialog wouldn't be visible so the method would get short circuited